WMF Framework/NavigationBar.swift

```diff
@@ -723,4 +706,14 @@ extension NavigationBar: UINavigationBarDelegate {
         delegate?.navigationController?.popViewController(animated: true)
         return false
     }
+
+    public func navigationBar(_ navigationBar: UINavigationBar, didPop item: UINavigationItem) {
```

Comment:
`shouldPop` still isn't called for a long-press pop in the latest beta which came out today (beta 5).

I think the weird animation is due to having to call this in `didPop` rather than `willPop`. (I've added some additional comments around this.) From slowing down the animations, it looks like the system pop is the weird artifact one, and then our manual pop is the second (real) one. I can't think of any way around this w/o being able to interrupt it in `willPop` - I'm very open to ideas though. I left a comment encouraging us to move it to `willPop` when that (presumed) bug gets fixed.

(Notably, `willPush` and `didPush` aren't called when pushing a new article on the stack. This could be because of how we push new articles. Potentially digging further into how we push new articles to the NavController could shed some light on other alternative ways to handle the popping. But that also feels like a potential rabbit hole - If this seems worthwhile to dig into in order to potentially improve this animation, let me know.)